### PR TITLE
Add Catalogs resource

### PR DESCRIPTION
**Pull request by julianmclain**
Hey Geoff - I'm using this wrapper in a project that needs to interact with the Catalogs endpoints. Do you mind if I add them? Should be able to get it done in the next week or so. I wanted to get a small PR in front of you to make sure things look good so far.

Thanks for your work btw!

## Proposed changes
- Ignore vscode config
- Small fixes in index.js and campaigns.test.js
- Add Catalogs resource and GET method

### CODE REVIEW

**Comment by geoffdutton** [comment]
@julianmclain Absolutely, looking good! Let me know when you're happy with it. Thanks!

**Comment by julianmclain** [comment]
Awesome will let you know. 1 quick question actually. Given that Catalog `items` and `fieldMappings` are implemented as nested resources underneath Catalog, do you have a preference on:

`iterable.catalogs.items.get({ catalogName, itemId })`

vs

`iterable.catalogs.getItem({ catalogName, itemId })`

My thought is the first option, but could see either way working well.

**Comment by geoffdutton** [comment]
Yeah I agree that `iterable.catalogs.items.get({ catalogName, itemId })` makes sense.

**Comment by julianmclain** [comment]
@geoffdutton Alrighty, lmk what you think. Based on the examples I setup it's is looking good.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = { "extends": "standard" };
```
Comment (geoffdutton) [inline .eslintrc.js]: Just curious... is this for IDEs/VScode?

File: index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const Iterable = require('./lib/iterable')
 module.exports = Iterable
 
 if (require.main === module) {
-  const client = Iterable('api key')
+  const client = Iterable.create('api key')
```
Comment (geoffdutton) [inline index.js]: Nice catch

File: .eslintrc.js (hunk 1 above)
Comment (julianmclain) [reply .eslintrc.js]: I added it so VScode would highlight / fix lint errors. I thought the purpose of an .eslintrc file is just setting custom linting rules, but I couldn't get VScode to catch errors without it. I can pop it out if you don't want to add it at this point.

Comment (geoffdutton) [reply .eslintrc.js]: Got it, no that's fine. Thanks.

**Comment by geoffdutton** [comment]
@julianmclain I published [v0.4.0](https://www.npmjs.com/package/node-iterable-api), thanks again for the contribution!

**Pull request merged**